src/esbmc/bmc.cpp

```diff
@@ -658,3 +688,58 @@ smt_convt::resultt bmct::run_thread(std::shared_ptr<symex_target_equationt> &eq)
     return smt_convt::P_ERROR;
   }
 }
+
+smt_convt::resultt bmct::multi_property_check(
+  std::shared_ptr<symex_target_equationt> &eq,
+  size_t remaining_claims)
+{
+  // As of now, it only makes sense to do this for the base-case
+  assert(
+    options.get_bool_option("base-case") &&
+    "Multi-property only supports base-case");
+  smt_convt::resultt final_result;
+
+  /* TODO: For coverage, We should store and remove all the claims that
+   *       were already verified
+   */
+  for(size_t i = 1; i <= remaining_claims; i++)
```

Comment:
don't we have a way to check all assertions at once with the SMT solvers?

Reply:
You mean using solvers API or running all of them in parallel? Either way, it should be possible. I would suggest this going in another PR though.

Reply:
Yes, I believe you should be able to do it via API.

I think this current solution is not good.

Reply:
@fbrausse and @mikhailramalho: what do you think?

Reply:
You can check the code in `symex_target_equationt::convert` to convert all the assertions at once, it should look something like:
```
    smt_conv.assert_ast(
      smt_conv.make_n_ary(&smt_conv, &smt_convt::mk_or, assertions));
```

Reply:
Given that @rafaelsamenezes has now parallelized the VCs check, I think we could consider @mikhailramalho's suggestions for a future PR.